### DISCUSSION TITLE
Auto-disconnect database connections

### DISCUSSION THREAD

**Discussion by johnmerritt72** [dbeaver/dbeaver#12398] [Q&A]
I use DBeaver at work, and our DBAs have a rule that says we must not leave idle connections open to our production database (DB2), especially overnight but even during the day for long periods of time.  I have looked for a way to automatically disconnect from the database after an elapsed period of time, but don't see an option for that.  Is this possible?

**Comment by uslss** [comment]
@johnmerritt72 mentioned option is available in Connection Settings
![image](https://user-images.githubusercontent.com/31996417/117799114-771b2300-b25a-11eb-81c1-dcfaacfaae59.png)

**Reply by johnmerritt72** [reply]
Thanks, I tried that but am still connected hours later, even after setting it to 60.  Is that setting for long running transactions only? I need to have it disconnect whether or not a transaction is running.

**Comment by guibhz** [comment]
Hi @johnmerritt72 , this problem is similar to the one I'm having (#12404 ). I'm trying to find a way to avoid keeping open idle connections, but unfortunatly, i think that there is no configuration to solve this problem.
One suggestion that I've found is this thread on StackOverflow (https://stackoverflow.com/questions/12391174/how-to-close-idle-connections-in-postgresql-automatically/30769511#30769511), where a user suggest a script to close iddle connections.

I didn't run this code because I don't have access to my production environment, but I will give it a shot in my development environment
